### PR TITLE
Show shortcut help screen when pressing `?`

### LINKED ISSUE #5230: Keyboard shortcuts for Piwik app
While introducing the ZenMode to Piwik in #4987 we have also started defining Keyboard shortcuts to make using Piwik app easier.

The goal of this ticket is to identify which actions could be available as shortcut, and which key to use for them.

Please comment with your suggestions and feedback!

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #5230

### CODE REVIEW

**Comment by tsteur** [comment]
I reckon making this extensible for plugins will be a different issue right? Could otherwise maybe just post an event with an array like `[z => {{ 'CoreHome_ShortcutZenMode'|translate }}]` and keep the template a bit more stupid? Some plugins add further shortcuts on some pages (such as session recording)

**Comment by mattab** [comment]
Would be good to make the Help popup already extensible if possible. 

Otherwise, nice new feature :+1:

**Comment by sgiehl** [comment]
As such shortcuts will always be added in JavaScript I'll try to move some logic there...

**Comment by tsteur** [comment]
👍

**Comment by sgiehl** [comment]
I've now implemented a new method `piwikHelper.registerShortcut`, which basically bypasses `Mousetrap.bind`, but additionally saves the keystroke and description for it, which will then be shown on the help screen...

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ The Product Changelog at **[piwik.org/changelog](https://piwik.org/changelog)**
 
 ### New Features
 * New config.ini.php setting `show_update_notification_to_superusers_only` makes it possible to hide update notifications for all users except of superusers
+* New JavaScript method `piwikHelper.registerShortcut` allows plugins to bind keyboard shortcuts. A summary for available shortcuts will be shown by pressing `?`
```
Comment (mattab) [inline CHANGELOG.md]: Let's move this mention into the 3.2.1 changelog section

File: core/Tracker/PageUrl.php [hunk 2]
```diff
@@ -174,7 +174,7 @@ protected static function cleanupHostAndHashTag($parsedUrl, $idSite = false)
         }
 
         if (!empty($parsedUrl['host'])) {
-            $parsedUrl['host'] = Common::mb_strtolower($parsedUrl['host'], 'UTF-8');
+            $parsedUrl['host'] = Common::mb_strtolower($parsedUrl['host']);
```
Comment (mattab) [inline core/Tracker/PageUrl.php]: This change is un-related and not for this PR?

Comment (sgiehl) [reply core/Tracker/PageUrl.php]: Yes. Unrelated. It removes an useless parameter as `Common::mb_strtolower` only takes one parameter

**Comment by mattab** [comment]
LGTM. @tsteur could you maybe check & merge this one?

**Comment by sgiehl** [comment]
Rebased branch and resolved conflicts... @tsteur any additional change requests?

**Pull request merged**